src/test/java/net/imagej/legacy/BatchModeTest.java

```diff
@@ -110,7 +111,18 @@ private void showLegacyUI() {
 		// TODO: This should be improved. The test should also work in headless mode, without the legacy ui visible.
 		assumeFalse(GraphicsEnvironment.isHeadless());
 		UIService service = context.service(UIService.class);
-		service.showUI();
+		Runnable showUI = new Runnable() {
+			@Override
+			public void run() {
+				service.showUI();
+			}
+		};
+		try {
+			EventQueue.invokeAndWait(showUI);
```

Comment:
Why not `context.service(ThreadService.class).invoke(...)`? I am concerned that the real bug-fix for these sorts of cases is to fix the `showUI()` behavior of AWT/Swing-based UIs to do things properly on the EDT in the first place, so that downstream consumers like this unit test don't have to worry about it. And if we eventually do that fix, then this sort of redundant wrapping will result in deadlocks.